src/java.base/share/classes/java/lang/invoke/InvokerBytecodeGenerator.java

```diff
@@ -1798,51 +1666,60 @@ static MemberName generateLambdaFormInterpreterEntryPoint(MethodType mt) {
     }
 
     private byte[] generateLambdaFormInterpreterEntryPointBytes() {
-        classFilePrologue();
-        methodPrologue();
-
-        // Suppress this method in backtraces displayed to the user.
-        mv.visitAnnotation(HIDDEN_SIG, true);
-
-        // Don't inline the interpreter entry.
-        mv.visitAnnotation(DONTINLINE_SIG, true);
-
-        // create parameter array
-        emitIconstInsn(invokerType.parameterCount());
-        mv.visitTypeInsn(Opcodes.ANEWARRAY, "java/lang/Object");
-
-        // fill parameter array
-        for (int i = 0; i < invokerType.parameterCount(); i++) {
-            Class<?> ptype = invokerType.parameterType(i);
-            mv.visitInsn(Opcodes.DUP);
-            emitIconstInsn(i);
-            emitLoadInsn(basicType(ptype), i);
-            // box if primitive type
-            if (ptype.isPrimitive()) {
-                emitBoxing(Wrapper.forPrimitiveType(ptype));
+        final byte[] classFile = classFilePrologue(new Consumer<ClassBuilder>() {
+            @Override
+            public void accept(ClassBuilder clb) {
+                methodPrologue(clb, new Consumer<MethodBuilder>() {
+                    @Override
+                    public void accept(MethodBuilder mb) {
+
+                        mb.with(RuntimeVisibleAnnotationsAttribute.of(List.of(
```

Comment:
Such shared annotation attributes are worth keeping elsewhere as a constant too. Unfortunately we aren't in Haskell, and constant expressions caching like JEP 303 will take a while.